client/question/services/FeedbackGeneratorService.js

```diff
@@ -24,18 +24,32 @@ tie.factory('FeedbackGeneratorService', [
   'RUNTIME_ERROR_FEEDBACK_MESSAGES', 'WRONG_LANGUAGE_ERRORS', 'LANGUAGE_PYTHON',
   'CLASS_NAME_AUXILIARY_CODE', 'CLASS_NAME_SYSTEM_CODE', 'PARAGRAPH_TYPE_TEXT',
   'PARAGRAPH_TYPE_CODE', 'PARAGRAPH_TYPE_SYNTAX_ERROR',
-  'PYTHON_PRIMER_BUTTON_NAME', 'ERROR_COUNTER_LANGUAGE_UNFAMILIARITY',
-  'ERROR_COUNTER_SAME_RUNTIME', 'UNFAMILIARITY_THRESHOLD',
-  'FEEDBACK_CATEGORIES',
+  'PARAGRAPH_TYPE_OUTPUT', 'PYTHON_PRIMER_BUTTON_NAME',
+  'ERROR_COUNTER_LANGUAGE_UNFAMILIARITY', 'ERROR_COUNTER_SAME_RUNTIME',
+  'UNFAMILIARITY_THRESHOLD', 'FEEDBACK_CATEGORIES', 'SAMPLE_INPUT',
+  'CORRECTNESS_TEST_INPUT_DISPLAYED',
+  'CORRECTNESS_TEST_EXPECTED_OUTPUT_DISPLAYED',
+  'CORRECTNESS_TEST_OBSERVED_OUTPUT_DISPLAYED', 'CORRECTNESS_FEEDBACK_TEXT',
   function(
     FeedbackObjectFactory, TranscriptService, ReinforcementGeneratorService,
     CODE_EXECUTION_TIMEOUT_SECONDS, SUPPORTED_PYTHON_LIBS,
     RUNTIME_ERROR_FEEDBACK_MESSAGES, WRONG_LANGUAGE_ERRORS, LANGUAGE_PYTHON,
     CLASS_NAME_AUXILIARY_CODE, CLASS_NAME_SYSTEM_CODE, PARAGRAPH_TYPE_TEXT,
     PARAGRAPH_TYPE_CODE, PARAGRAPH_TYPE_SYNTAX_ERROR,
-    PYTHON_PRIMER_BUTTON_NAME, ERROR_COUNTER_LANGUAGE_UNFAMILIARITY,
-    ERROR_COUNTER_SAME_RUNTIME, UNFAMILIARITY_THRESHOLD,
-    FEEDBACK_CATEGORIES) {
+    PARAGRAPH_TYPE_OUTPUT, PYTHON_PRIMER_BUTTON_NAME,
+    ERROR_COUNTER_LANGUAGE_UNFAMILIARITY, ERROR_COUNTER_SAME_RUNTIME,
+    UNFAMILIARITY_THRESHOLD, FEEDBACK_CATEGORIES, SAMPLE_INPUT,
+    CORRECTNESS_TEST_INPUT_DISPLAYED,
+    CORRECTNESS_TEST_EXPECTED_OUTPUT_DISPLAYED,
+    CORRECTNESS_TEST_OBSERVED_OUTPUT_DISPLAYED, CORRECTNESS_FEEDBACK_TEXT) {
+
+    /**
+     * Object used to keep track of how many correctness feedback (values)
```

Comment:
I thought we decided that this was going to have values that were strings representing states, as opposed to numeric counts?

Reply:
We are. I just forgot to update the variable description. Fixed.